gcc/rust/hir/rust-ast-lower-pattern.cc

```diff
@@ -333,9 +333,12 @@ ASTLoweringPattern::visit (AST::AltPattern &pattern)
     = new HIR::AltPattern (mapping, std::move (alts), pattern.get_locus ());
 
   if (is_let_top_level)
-    rust_error_at (pattern.get_locus (),
-		   "top level alternate patterns are not allowed for %<let%> "
-		   "bindings - use an outer grouped pattern");
+    {
+      rich_location richloc (line_table, pattern.get_locus ());
+      richloc.add_fixit_replace ("wrap the pattern in parentheses: ( _ | _ )");
```

Comment:
This is not true for Rust 1.49, which we are targeting.
```suggestion
```

```
arthur@platypus /tmp [1]> cat main.rs
fn main() {
    let (a | a) = 15;
}
arthur@platypus /tmp> rustc --version
rustc 1.49.0 (e1884a8e3 2020-12-29)
arthur@platypus /tmp> rustc main.rs
error[E0658]: or-patterns syntax is experimental
 --> main.rs:2:10
  |
2 |     let (a | a) = 15;
  |          ^^^^^
  |
  = note: see issue #54883 <https://github.com/rust-lang/rust/issues/54883> for more information

warning: unnecessary parentheses around pattern
 --> main.rs:2:9
  |
2 |     let (a | a) = 15;
  |         ^^^^^^^ help: remove these parentheses
  |
  = note: `#[warn(unused_parens)]` on by default

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0658`.
```

Reply:
Thanks for review :)